### PR TITLE
Don't let KHRONOS_STATIC affect the calling convention

### PR DESCRIPTION
**Pull request by mstorsjo**
When intending to link statically, we should leave out
__declspec(dllimport), but the calling convention (__stdcall) should
still be the same.

This reverts parts of commit 94ba8ee876206364cf45a9bc08b8db5a52cb9543,
making the exception to __stdcall specific to __SCITECH_SNAP__ again
as it was before.

### CODE REVIEW

**Comment by stonesthrow** [comment]
Reference: https://github.com/KhronosGroup/EGL-Registry/pull/81

**Comment by mstorsjo** [comment]
Thanks!

Well, with the previous change, if they applied `KHRONOS_STATIC` consistently both when building libEGL and the caller of it, it shouldn't change anything (it will use a different calling convention but both change in sync).

If libEGL and the caller didn't both use `KHRONOS_STATIC` (but was built for static linking by other means, e.g. the build of libEGL did `-DEGLAPI=`, which happens e.g. when ANGLE is built as part of Qt), it's currently impossible to link against it using `KHRONOS_STATIC`.

So in short, cases that worked before should still work, and it should fix mismatches in other cases.

**Comment by stonesthrow** [comment]
Can be closed.
